sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_settings_client.py

```diff
@@ -72,6 +72,17 @@ def update_setting(self, setting: KeyVaultSetting, **kwargs) -> KeyVaultSetting:
         :rtype: ~azure.keyvault.administration.KeyVaultSetting
         :raises: :class:`~azure.core.exceptions.HttpResponseError`
         """
+        # Challenge authentication doesn't work as expected with this method's request endpoint
+        # If we need to prompt an auth challenge, do so using `get_setting` instead
+        # (Note: we prompt a challenge instead of fetching a token directly so the token has correct permissions)
+        challenge_cached = HttpChallengeCache.get_challenge_for_url(self._vault_url)
+        if not challenge_cached:
+            # `get_setting` is just used to prompt a challenge; we don't want to stop on any errors it may raise
+            try:
+                self._client.get_setting(vault_base_url=self._vault_url, setting_name=setting.name, **kwargs)
+            except Exception:  # pylint:disable=broad-except
+                pass
+
```

Comment:
Is there a tracking bug open on this to remove this extra request once fixed service-side?

Reply:
I'm tracking that with this issue (which is linked to this PR but won't be resolved by a merge): https://github.com/Azure/azure-sdk-for-python/issues/30382

As for whether to leave it up to the customer: I see your point 🙂 I doubt many people (if any) are using the settings API with Python, and the service team acknowledged the issue and intend to fix it. But if I don't fix it here the CLI might implement the workaround on their own -- even if they don't, I don't think there's much downside to us providing the fix for a preview version.